internal/ethapi/api.coreth_test.go

```diff
@@ -0,0 +1,159 @@
+// (c) 2025, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package ethapi
+
+import (
+	"context"
+	"math/big"
+	"testing"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/common/hexutil"
+	"github.com/stretchr/testify/require"
+)
+
+type testSuggestPriceOptionsBackend struct {
+	Backend // embeds the interface to avoid implementing unused methods
+
+	estimateBaseFee  *big.Int
+	suggestGasTipCap *big.Int
+}
+
+func (b *testSuggestPriceOptionsBackend) EstimateBaseFee(context.Context) (*big.Int, error) {
+	return b.estimateBaseFee, nil
+}
+
+func (b *testSuggestPriceOptionsBackend) SuggestGasTipCap(context.Context) (*big.Int, error) {
+	return b.suggestGasTipCap, nil
+}
+
+func TestSuggestPriceOptions(t *testing.T) {
+	tests := []struct {
+		name             string
+		estimateBaseFee  *big.Int
+		suggestGasTipCap *big.Int
+		expected         *PriceOptions
+	}{
+		{
+			name:             "nil_base_fee",
+			estimateBaseFee:  nil,
+			suggestGasTipCap: common.Big1,
+			expected:         nil,
+		},
+		{
+			name:             "nil_tip_cap",
+			estimateBaseFee:  common.Big1,
+			suggestGasTipCap: nil,
+			expected:         nil,
+		},
+		{
+			name:             "minimum_values",
+			estimateBaseFee:  bigMinBaseFee,
+			suggestGasTipCap: bigMinGasTip,
+			expected: &PriceOptions{
+				Slow: &Price{
```

Comment:
nit: might be useful to make a helper function to create Price like `newPrice(gasTip, gasFee)` to avoid repeating `(*hexutil.Big)(big.NewInt(...))` in this test.
or just one for `(*hexutil.Big)(big.NewInt(...))` but I'm not sure what to call that